### PR TITLE
SFR-1504: Add cucumber and playwright

### PR DESCRIPTION
**Pull request by samanthaandrews**
[Jira Ticket](http://jira.nypl.org/browse/SFR-1504)

### This PR does the following:
- Sets up a very basic Cucumber and Playwright configuration for running end-to-end tests

### Testing requirements & instructions: 
-  Checkout this branch and run `npm run cucumber` and a simple Playwright test should pass

### CODE REVIEW

File: playwright/support/setup.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  Page,
+} from "@playwright/test";
+
+interface PlaywrightParameters {
+  browser: Browser;
+  context: BrowserContext;
+  page: Page;
+}
+
+export class CustomWorld extends World {
```
Comment (hokei) [inline playwright/support/setup.ts:23]: Can you explain a bit what's going on in this file (or share a link to docs)? A "World" object seems like a strange thing in the JS context.

Comment (samanthaandrews) [reply playwright/support/setup.ts:23]: Sure thing. This is a link to the [cucumber-js docs](https://github.com/cucumber/cucumber-js/blob/main/docs/support_files/world.md) where it talks about the World scope. It's basically a Cucumber setup file. In our case, we need to extend the default cucumber-js world to include Playwright-specific browser objects.

**Comment by samanthaandrews** [comment]
@hokei @jackiequach I told Chris to hold off making any commits/pushes to this branch in case we want to merge it to the repo as it is currently. Let me know how you'd like to proceed!

**Comment by chrismulholland** [comment]
This was successful for me:  "Checkout this branch and run npm run cucumber and a simple Playwright test should pass".  thanks @samanthaandrews!

**Review by jackiequach — APPROVED** [review]
looks good! just merge in dev

**Pull request merged**